server/hedley/modules/custom/hedley_reports/hedley_reports.module

```diff
@@ -3416,6 +3407,969 @@ function hedley_reports_count_acute_illness_signs(array $values) {
+ *
+ * This function checks whether the first dose of OPV (Oral Polio Vaccine) was
+ * given within 14 days of the child's birth, based on the vaccination progress
+ * data.
+ *
+ * @param array $encounter_data
+ *   An associative array containing details of the encounter, including
+ *   vaccination progress.
+ * @param DateTime $birth_date_obj
+ *   The DateTime object representing the child's birth date.
+ *
+ * @return bool
+ *   TRUE if the initial OPV dose was administered within 14 days of birth,
+ *   FALSE otherwise.
+ */
+function hedley_reports_well_child_initial_opv_administered(array $encounter_data, DateTime $birth_date_obj) {
+  $vaccination_progress = $encounter_data['vaccination_progress'];
+
+  if (empty($vaccination_progress['opv']) || empty($vaccination_progress['opv']['dose-1'])) {
+    return FALSE;
+  }
+
+  $first_dose_date_obj = new DateTime($vaccination_progress['opv']['dose-1']);
+  $interval = $first_dose_date_obj->diff($birth_date_obj);
+
+  return $interval->days < 14;
+}
+
+/**
+ * Retrieves the last expected dose for a specific vaccine type.
+ *
+ * This function returns the last dose that should be administered for a given
+ * initial vaccine type, based on the child's vaccination history and whether
+ * the OPV dose was administered.
+ *
+ * @param string $immunisation_type
+ *   The type of immunization (e.g., 'bcg', 'opv', 'dtp').
+ * @param bool $initial_opv_administered
+ *   Indicates whether the initial OPV dose was administered within 14 days
+ *   of birth.
+ *
+ * @return string|bool
+ *   The last expected dose for the vaccine (e.g., 'dose-1', 'dose-3'),
+ *   or FALSE, if the immunization type is not recognized.
+ */
+function hedley_reports_well_child_get_last_dose_for_vaccine($immunisation_type, $initial_opv_administered) {
+  switch ($immunisation_type) {
+    case 'bcg':
+    case 'ipv':
+    case 'dtp_sa':
+      return 'dose-1';
+
+    case 'rotarix':
+    case 'mr':
+    case 'hpv':
+      return 'dose-2';
+
+    case 'dtp':
+    case 'pcv13':
+      return 'dose-3';
+
+    case 'opv':
+      return $initial_opv_administered ? 'dose-4' : 'dose-3';
+  }
+
+  return FALSE;
+}
+
+/**
+ * Retrieves the interval between doses for a specific vaccine type.
+ *
+ * This function returns the time interval that should be observed between
+ * doses for a given vaccine, taking into account the site where the encounter
+ * occurs.
+ *
+ * @param string $immunisation_type
+ *   The type of immunization (e.g., 'bcg', 'opv', 'dtp').
+ * @param string $site
+ *   The site name, which may affect the interval for certain vaccines.
+ *
+ * @return string|bool
+ *   A string representing the interval between doses (e.g., '+4 weeks',
+ *   '+9 months'), or FALSE if the immunization type is not recognized.
+ */
+function hedley_reports_well_child_get_interval_for_vaccine($immunisation_type, $site) {
+  switch ($immunisation_type) {
+    case 'bcg':
+    case 'ipv':
+    case 'dtp_sa':
+      return '+0 days';
```

Comment:
Does this mean that it's a one-time shot?

Reply:
Yep